rust/vectordb/src/table.rs

```diff
@@ -69,25 +63,42 @@ impl Table {
     ///
     /// * `base_path` - The base path where the table is located
     /// * `name` The Table name
-    /// * `params` The [OpenTableParams] to use when opening the table
+    /// * `params` The [ReadParams] to use when opening the table
     ///
     /// # Returns
     ///
     /// * A [Table] object.
-    pub async fn open_with_params(
-        base_uri: &str,
-        name: &str,
-        params: OpenTableParams,
-    ) -> Result<Self> {
-        let path = Path::new(base_uri);
+    pub async fn open_with_params(uri: &str, name: &str, params: &ReadParams) -> Result<Self> {
```

Comment:
do we still need `name`?

Reply:
Yea, this is for the table to have a better display name without leaking storage details. 

Might be able to deduct the name from uri if not provided, wdyt @chebbyChefNEQ ?